src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -1830,4 +1837,33 @@ protected function db_errors_happened() {
 
 		return $wpdb->last_error !== $this->last_wpdb_error;
 	}
+
+	/**
+	 * Cast error responses from the Service to WP_Errors to ease processing down the line.
+	 *
+	 * If a response is a WP_Error already or is not an error response then it will not be modified.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Error|object $import_data
+	 *
+	 * @return array|\WP_Error
+	 */
+	protected function maybe_cast_to_error( $import_data ) {
+		if ( is_wp_error( $import_data ) ) {
+			return $import_data;
+		}
+
+		if ( ! empty( $import_data->status ) && 'error' === $import_data->status ) {
```

Comment:
I wonder if we should _also_ look and see if the response was something other than `200` and generate an error as well. Right now with our throttling, we return `403`.